src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -21,35 +47,378 @@ const ActorInit En_Kakasi3_InitVars = {
+            this->unk_19C[3] = (s16)Math_Rand_ZeroFloat(10.99f) + 30;
+            this->unk_1A6 = (s16)Math_Rand_ZeroFloat(4.99f);
+        }
+
+        this->unk_19A = (s16)Math_Rand_ZeroFloat(2.99f) + 5;
+        phi_v0 = this->unk_1A6;
+    }
+    switch (phi_v0) {
+        case 0:
+            this->unk_19A++;
+            if (this->unk_1A4 == 0) {
+                this->unk_1A4 = 1;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_ROLL);
+            }
+            break;
+        case 1:
+            this->unk_19A++;
+            this->unk_1B8 = 1.0f;
+            break;
+        case 2:
+            this->unk_19A++;
+            if (this->unk_1AE == 0) {
+                this->unk_1AE = 5000;
+            }
+            break;
+        case 3:
+            this->unk_19A++;
+            if (this->unk_1AA == 0) {
+                this->unk_1AA = 5000;
+            }
+            break;
+        case 4:
+            this->unk_19A++;
+            this->unk_1B8 = 2.0f;
+            break;
+    }
+
+    if (this->unk_19A > 8) {
+        this->unk_19A = 8;
+    }
+
+    if (this->unk_19A != 0) {
+        this->actor.gravity = -1.0f;
+        if (this->unk_19A == 8 && (this->actor.bgCheckFlags & 1)) {
+            this->actor.velocity.y = 3.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_IT_KAKASHI_JUMP);
+        }
+        Math_SmoothScaleMaxF(&this->skelAnime.animPlaybackSpeed, this->unk_1B8, 0.1, 0.2);
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.x, this->unk_1AA, 5, 1000, 0);
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.z, this->unk_1AE, 5, 1000, 0);
+
+        if (this->unk_1AA != 0 && fabsf(this->actor.shape.rot.x - this->unk_1AA) < 50.0f) {
+            this->unk_1AA *= -1.0f;
+        }
+        if (this->unk_1AE != 0 && fabsf(this->actor.shape.rot.z - this->unk_1AE) < 50.0f) {
+            this->unk_1AE *= -1.0f;
+        }
+
+        if (this->unk_1A4 != 0) {
+            this->actor.shape.rot.y += 0x1000;
+            if (this->actor.shape.rot.y == 0) {
+                this->unk_1A4 = 0;
+            }
+        }
+        currentFrame = this->skelAnime.animCurrentFrame;
+        if (currentFrame == 11 || currentFrame == 17) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_SWING);
+        }
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+}
+
+void func_80A911F0(EnKakasi3* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000214.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000214, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
+    this->actionFunc = func_80A91284;
+}
+
+void func_80A91284(EnKakasi3* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    this->actor.textId = 0x40A1;
+    this->dialogState = 6;
+    this->unk_19A = 0;
+
+    if (LINK_IS_CHILD) {
+        this->unk_194 = 0;
+        if (gSaveContext.unk_12C5) {
```

Comment:
Based on existing usage I don't think we can really tell whether this is a boolean.
```suggestion
        if (gSaveContext.unk_12C5 != 0) {
```